### ISSUE TITLE
[Appium 1.8.x+] Native tap not working on iOS real device (Unhandled endpoint ... /wda/touch/perform)

### ISSUE THREAD

**Issue by SealviewTW** [appium/appium#11275]
TouchAction (native tap) does not work since I've updated Appium from 1.7.X to 1.8.X+ (this includes latest 1.9.0 release).

Error: https://gist.github.com/SealviewTW/a29a4966cb2e41c96e3595da324987ab

* Appium version (or git revision) that exhibits the issue: 1.8.X up to 1.9.0
* Last Appium version that did not exhibit the issue (if applicable): 1.7.2
* Desktop OS/version used to run Appium: macOS High Sierra 10.13.6
* Node.js version (unless using Appium.app|exe): 6.3.0
* Mobile platform/version under test: iOS v11.4.1
* Real device or emulator/simulator: Real device (iPad Air 2)

Appium server logs: https://gist.github.com/SealviewTW/877ca1c0f7b0bba6657eb7395d683568

Code To Reproduce Issue: https://gist.github.com/SealviewTW/a8be67779e98b32570e8da8e67412026

**Comment by mykola-mokhnach** [comment]
You must be in NATIVE context in order to perform touch actions

**Comment by mykola-mokhnach** [comment]
Also make sure the WDA binary on the device is up to date (it is cached by default)

**Comment by SealviewTW** [comment]
The WDA was freshly installed after updating to 1.9.0

As for the NATIVE context, please help me understand, but are these calls not a successful (Response code 200) switch to native?

2018-08-28 14:49:12:787 - debug: [MJSONWP] Calling AppiumDriver.setContext() with args: ["NATIVE_APP","90ee2da0-0b90-4504-85ca-aad7de0cde9f"]
2018-08-28 14:49:12:787 - debug: [XCUITest] Executing command 'setContext'
2018-08-28 14:49:12:787 - debug: [iOS] Attempting to set context to 'NATIVE_APP'
2018-08-28 14:49:12:787 - debug: [RemoteDebugger] Disconnecting from WebKit remote debugger
2018-08-28 14:49:12:788 - debug: [MJSONWP] Responding to client with driver.setContext() result: null
2018-08-28 14:49:12:789 - info: [HTTP] <-- POST /wd/hub/session/90ee2da0-0b90-4504-85ca-aad7de0cde9f/context 200 2 ms - 76
2018-08-28 14:49:12:789 - info: [HTTP] 
2018-08-28 14:49:12:789 - debug: [RemoteDebugger] WebKit remote debugger socket disconnected

**Comment by mykola-mokhnach** [comment]
> The WDA was freshly installed after updating to 1.9.0

Only the driver was updated, but the binary might stay cached. Remove it from the device manually and clear the xcode caches

**Comment by mykola-mokhnach** [comment]
> "time": "May 18 2018 11:51:46"

Also the WDA build time does not look very promising

**Comment by SealviewTW** [comment]
Thank you for your help.

I have managed to fix my issue by doing the following steps:

1. Uninstall appium
2. Restart mac
3. Fresh install appium
4. Clean + Rebuild wda (and this was probably the fault, RESIGN BOTH WebDrvierAgentLib and WebDriverAgentRunner modules inside the xcodeproj before clean + build)
5. Uninstall WDA from device
6. Restart device

Closing the issue as it was due to improper version WDA
